Sources/ProjectSpec/AggregateTarget.swift

```diff
@@ -0,0 +1,63 @@
+import Foundation
+import JSONUtilities
+
+public struct AggregateTarget: ProjectTarget {
+    public var name: String
+    public var targets: [String]
+    public var settings: Settings
+    public var buildScripts: [BuildScript]
+    public var configFiles: [String: String]
+    public var scheme: TargetScheme?
+    public var attributes: [String: Any]
+
+    public init(
+        name: String,
+        targets: [String],
+        settings: Settings = .empty,
+        configFiles: [String: String] = [:],
+        buildScripts: [BuildScript] = [],
+        scheme: TargetScheme? = nil,
+        attributes: [String: Any] = [:]
+    ) {
+        self.name = name
+        self.targets = targets
+        self.settings = settings
+        self.configFiles = configFiles
+        self.buildScripts = buildScripts
+        self.scheme = scheme
+        self.attributes = attributes
+    }
+}
+
+extension AggregateTarget: CustomStringConvertible {
+
+    public var description: String {
+        return "\(name): \(targets.joined(separator: ", "))"
```

Comment:
When `targets` is empty this shouldn't include `:`. It's valid to not have any targets if it's only used to run a script, like `carthage-verify.sh`.

Reply:
Yeah at the moment `targets` is a required field. Are there cases where you wouldn't actually aggregate targets? Isn't that the point of them? I haven't really used them before so I'm not sure

Reply:
Yes. They are named weird in that way. It's also a common way to just run scripts. We use it to `carthage-verify.sh` before all our other targets so we get a fast failure and don't have to run the same script for every target in our project.

Reply:
Ok cool. A normal empty target would achieve the same thing right, this is just better suited to it?

Reply:
An empty target fails to build I believe, though I haven't tried that hard since I've only really done it this way (which I believe is the recommended way).